lsm/src/main/java/org/apache/iotdb/lsm/manager/LsmManager.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iotdb.lsm.manager;
+
+import org.apache.iotdb.lsm.context.RequestContext;
+import org.apache.iotdb.lsm.levelProcess.LevelProcess;
+
+// used to implement lsm manager
+public interface LsmManager<T, C extends RequestContext> {
+
+  /**
+   * use this method to process root memory node
+   *
+   * @param memNode memory node
+   * @param context request context
+   */
+  void process(T memNode, C context) throws Exception;
+
+  /**
+   * add the LevelProcess of the next layer of memory nodes
+   *
+   * @param next LevelProcess of the next layer
+   * @return LevelProcess of the next layer
+   */
+  <O> LevelProcess<T, O, C> nextLevel(LevelProcess<T, O, C> next);
```

Comment:
It seems this method is used to build the process method of ```LSMManager```. What about separating these building methods into an individual  ```LSMManagerBuilder```. The ```LSMManager``` will only keep the executing methods. PLZ refer to the design pattern, Builder Pattern.